### ISSUE TITLE
Acquire::http::Proxy "https://proxy:443/"; or: https to proxy?

### ISSUE THREAD

**Issue by chiasamen** [Debian/apt#42]
Problem is: I have a proxy which only accepts https.
Is it possible to speak https to a proxy?
How could I workaround if not possible?

I tried:
```
Acquire::http::Proxy "https://proxy:443/";
Acquire::https::Proxy "https://proxy:443/";
```
But I got:

>Err:3 https://security.ubuntu.com/ubuntu xenial-security Release
>  Recv failure: Connection reset by peer

Which seem to be the issue if the proxy gets http on a https-port

**Comment by julian-klode** [comment]
Nö, it's not possible right now. I think you could use stunnel to provide a local http endpoint.

**Comment by julian-klode** [comment]
Fixed in 1.5
